### PR TITLE
Fix HystrixCommandInterceptor

### PR DESCRIPTION
**Pull request by mkouba**
- each intercepted method must have a unique command key
- also simplify tests and few minor changes

### CODE REVIEW

File: src/main/java/org/wildfly/swarm/microprofile/fault/tolerance/hystrix/HystrixCommandInterceptor.java [hunk 1]
```diff
@@ -47,33 +47,30 @@
...
+        // TODO
+        // CircuitBreaker circuitBreaker = getAnnotation(method, CircuitBreaker.class);
+        // Retry retry = getAnnotation(method, Retry.class);
+
+        Fallback fallback = getAnnotation(method, Fallback.class);
+        Supplier<Object> fallbackToRun = null;
 
         if (fallback != null) {
-            FallbackHandler fbh = fallback.value().newInstance();
+            FallbackHandler<?> fbh = fallback.value().newInstance();
```
Comment (manovotn) [inline src/main/java/org/wildfly/swarm/microprofile/fault/tolerance/hystrix/HystrixCommandInterceptor.java]: Just a quick thought - in order to invoke `newInstance()`, there has to an accessible no-args constructor of this fallback class. The MP spec at it's current state does not mandate this. Could it be a problem?

Comment (mkouba) [inline src/main/java/org/wildfly/swarm/microprofile/fault/tolerance/hystrix/HystrixCommandInterceptor.java]: I think this part will be completely rewritten. See also https://github.com/eclipse/microprofile-fault-tolerance/issues/4#issuecomment-308676325

File: src/main/java/org/wildfly/swarm/microprofile/fault/tolerance/hystrix/HystrixCommandInterceptor.java [hunk 2]
```diff
@@ -82,6 +79,19 @@ public Object interceptCommand(InvocationContext ic) throws Exception {
...
+
+        Timeout timeout = getAnnotation(method, Timeout.class);
+        if (timeout != null) {
+            // TODO: In theory a user might specify a long value
+            propertiesSetter.withExecutionTimeoutInMilliseconds((int) Duration.of(timeout.value(), timeout.unit()).toMillis());
+        }
+
+        return Setter.withGroupKey(HystrixCommandGroupKey.Factory.asKey("DefaultCommandGroup"))
+                // Each method has a unique command key
+                .andCommandKey(HystrixCommandKey.Factory.asKey(method.toString())).andCommandPropertiesDefaults(propertiesSetter);
```
Comment (manovotn) [inline src/main/java/org/wildfly/swarm/microprofile/fault/tolerance/hystrix/HystrixCommandInterceptor.java]: If you need unique for each method, is `method.toString()` enough?
How about overloaded methods or methods with same name in multiple classes?

Comment (mkouba) [inline src/main/java/org/wildfly/swarm/microprofile/fault/tolerance/hystrix/HystrixCommandInterceptor.java]: Yep, you're right, I thought `Method.toString()` includes declaring class which is not the case. I'm going to add declaring class.

**Comment by mkouba** [comment]
Please don't merge yet. I'm going to change the fallback logic to align with the MP.

**Comment by mkouba** [comment]
It's ready. Fallback handler lifecycle is aligned but validation is not implemented yet.

**Comment by mkouba** [comment]
> could be great if API would allow doing the same for the command

I'm not sure what you mean. Maybe create a MP issue?

**Comment by antoinesd** [comment]
> > could be great if API would allow doing the same for the command
> 
> I'm not sure what you mean. Maybe create a MP issue?

Yes MP FT API should provide an annotation like `@Fallback` to provide a class for normal execution. It would allow our users to create their own implementation of `DefaultCommand`

**Comment by antoinesd** [comment]
@mkouba Your PR is good for me. Is it ready for merging?

**Comment by mkouba** [comment]
@antoinesd Yes, it's ready.

**Pull request merged**